Backend/BackendAPI/Controllers/BikesController.cs

```diff
@@ -0,0 +1,110 @@
+﻿using BackendAPI.Models;
+using BackendAPI.Helpers.DTOFactories;
+using BackendAPI.Repository.Interfaces;
+using ClassLibrary.DTO;
+using ClassLibrary.Exceptions;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace BackendAPI.Controllers
+{
+    [Route("api/[controller]")]
+    [Authorize]
+    [ApiController]
+    public class BikesController : ControllerBase
+    {
+        private int GetRequestingUserID => int.Parse(
+                User.FindFirstValue(ClaimTypes.NameIdentifier));
+
+        private IRentalRepository rentalRepository;
+        private IBikeRepository bikeRepository;
+        private IUserRepository userRepository;
+
+        public BikesController(IRentalRepository rentalRepo, IBikeRepository bikeRepo, IUserRepository userRepo)
+        {
+            rentalRepository = rentalRepo;
+            bikeRepository = bikeRepo;
+            userRepository = userRepo;
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        [HttpGet("rented")]
+        public ActionResult RentedGet()
+        {
+            int userId = GetRequestingUserID;
+
+            var rentedBikes = rentalRepository.FindActiveRentals(userId)
+                .Select(r => BikeDTOFactory.CreateBikeDTO(r.Bike, r.User));
+
+            return Ok(new { Bikes = rentedBikes });
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        [HttpPost("rented")]
+        public ActionResult<BikeDTO> RentedPost([FromBody] IdDTO rent)
+        {
+            //[BLOCKED] dopisac check na zablokowanego uzytkownika
+
+            if (!int.TryParse(rent.Id, out int bikeId))
+                throw new HttpResponseException("Wrong Bike Id", 404);
```

Comment:
Ja wiem że tego nie ma w specce ale jak już to dopiszą to pewnie będzie niezgodnie. To zwraca jsona o polu "Error" a według ramla wszędzie gdzie piszą o 404 chcemy jsona o polu "messsage" (typ ErrorDTO). Ja ogólnie pomijałem TryParse i robiłem po prostu robiłem Parse, więc jeśli wolisz TryParse to daj znać to ja sie dostosuję i pozmieniam (może faktycznie TryParse jest lepsze). W każdym razie tutaj po prostu bardziej przyszłościowo lepiej zwrócić NotFoundObjectResult(new ErrorDTO(){Message = "Bike not found"}) (zakładam, że kiedyś dopiszą w specyfikacji tutaj 404)